src/intellisense/langIntellisenseProvider.ts

```diff
@@ -646,21 +653,52 @@ class ReferenceCompletionProvider extends IntellisenseProvider implements vscode
                     return labels.map(label => new vscode.CompletionItem(label, vscode.CompletionItemKind.Reference));
                 }).flat();
             case 1:
-                const bibUri = vfs.pathToUri(`${OUTPUT_FOLDER_NAME}/output.bbl`);
-                const content = new TextDecoder().decode( await vfs.openFile(bibUri) );
-                const regex = /\\bibitem\{([^\}]*)\}/g;
                 const items = new Array<vscode.CompletionItem>();
-                let match: RegExpExecArray | null;
-                while (match = regex.exec(content)) {
-                    const item = new vscode.CompletionItem(match[1], vscode.CompletionItemKind.Reference);
-                    items.push(item);
+                items.push(... await this.getBibCompletionItems(vfs, this.symbolProvider?.getBibList() ?? []));
+                if ( 
+                    vfs.walk((entity) => {
```

Comment:
move this out.

Reply:
Can you help explain why you need to walk the who root to find "/.outputs"?

Reply:
Explanation, please

Reply:
This is simply replace the used `list` to `walk` api. The reason to search the existence of `.bbl` file is to make sure this process is not terminated by openning the unexist .bbl file.

Reply:
Actually, it is might be a better choice to delete this part -- extraction from bbl file.

Reply:
I am asking why? If the path is fixed, why bother ***search*** for it? You open it, or you stat it, and you catch the error. Why not?